packages/backend/index.js

```diff
@@ -40,8 +40,33 @@ app.get("/sign-message", async (req, res) => {
   const messageId = req.query.messageId;
   const options = req.query;
 
+  // necessary to cast into JSON otherwise it is a string and verifySignature fails (because the objects are different)
+  const transformedObject = {
+    ...options,
+    batch: (() => {
+      if (!options.batch) return undefined;
+
+      // Parse if batch is a string
+      const parsedBatch = typeof options.batch === "string" ? JSON.parse(options.batch) : options.batch;
+
+      // Create a new object and include only defined properties
+      const resultBatch = {};
+      if (parsedBatch.number !== undefined) {
+        resultBatch.number = parsedBatch.number;
+      }
+      if (parsedBatch.status !== undefined) {
+        resultBatch.status = parsedBatch.status;
+      }
+
+      // Return resultBatch only if it has properties, otherwise return undefined
+      return Object.keys(resultBatch).length > 0 ? resultBatch : undefined;
+    })(),
+  };
```

Comment:
I have the feeling that passing `batch` as an object is overcomplicating everything (this transform, how you manage state in the frontend, destructuring in the backend, etc). 

How would everything look like if we had two separate variables? (eg `batchNumber`, `batchStatus`)?

I think it makes more sense, since most of the time we'd only want to change the status.

Reply:
Solved :)